model/QueueBroker/NewSqlQueueBroker.php

```diff
@@ -0,0 +1,267 @@
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoTaskQueue\model\QueueBroker;
+
+use common_persistence_Manager;
+use common_persistence_SqlPersistence;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\ParameterType;
+use Doctrine\DBAL\Query\QueryBuilder;
+use Doctrine\DBAL\Schema\AbstractSchemaManager;
+use Exception;
+use InvalidArgumentException;
+use oat\generis\Helper\UuidPrimaryKeyTrait;
+use oat\tao\model\taskQueue\Queue\Broker\AbstractQueueBroker;
+use oat\tao\model\taskQueue\Task\TaskInterface;
+use oat\taoTaskQueue\model\QueueBroker\storage\NewSqlSchema;
+use PDO;
+use Throwable;
+
+/**
+ * Storing messages/tasks in newSQl DB.
+ */
+class NewSqlQueueBroker extends AbstractQueueBroker
+{
+    use UuidPrimaryKeyTrait;
+
+    private $persistenceId;
+
+    /** @var common_persistence_SqlPersistence */
+    protected $persistence;
+
+    public function __construct(string $persistenceId, int $receiveTasks = 1)
+    {
+        parent::__construct($receiveTasks);
+
+        if (empty($persistenceId)) {
+            throw new InvalidArgumentException("Persistence id needs to be set for " . __CLASS__);
+        }
+
+        $this->persistenceId = $persistenceId;
+    }
+
+    public function __toPhpCode()
+    {
+        return 'new ' . get_called_class() . '('
+            . \common_Utils::toHumanReadablePhpString($this->persistenceId)
+            . ', '
+            . \common_Utils::toHumanReadablePhpString($this->getNumberOfTasksToReceive())
+            . ')';
+    }
+
+    /**
+     * Note: this method can be run multiple times because only the migrate queries
+     * (result of getMigrateSchemaSql) will be run.
+     *
+     * @inheritdoc
+     */
+    public function createQueue()
+    {
+        $persistence = $this->getPersistence();
+        /** @var AbstractSchemaManager $schemaManager */
+        $schemaManager = $persistence->getDriver()->getSchemaManager();
+        $schema = $schemaManager->createSchema();
+        $fromSchema = clone $schema;
+        try {
+            $schema->dropTable($this->getTableName());
+        } catch (Throwable $exception) {
+            $this->logDebug('Schema of ' . $this->getTableName() . ' table already up to date.');
+        }
+        // Create the table
+        $schema = $this->getSchemaProvider()
+            ->setQueueName($this->getQueueName())
+            ->getSchema($schema, $this->getTableName());
+        $queries = $persistence->getPlatForm()->getMigrateSchemaSql($fromSchema, $schema);
+        foreach ($queries as $query) {
+            $persistence->exec($query);
+        }
+    }
+
+    /**
+     * Insert a new task into the queue table.
+     */
+    public function push(TaskInterface $task)
```

Comment:
```suggestion
    public function push(TaskInterface $task): bool
```
Any reason we should avoid  return type  specification?

Reply:
done